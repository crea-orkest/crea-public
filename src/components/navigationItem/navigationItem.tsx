import Link from 'next/link'
import React from 'react'
import classNames from 'classnames'
import { slugFormatter } from 'utils/slugFormatter'
import styles from './styles.module.scss'

export interface NavigationItemProps {
  slug?: string | null
  label?: string | null
}

export const NavigationItem: React.FC<NavigationItemProps> = ({
  slug,
  label,
}) => {
  if (!slug)
    return <li className={classNames(styles.navItem)}>No link specified</li>
  if (!label)
    return <li className={classNames(styles.navItem)}>No label specified</li>

  return (
    <li className={classNames(styles.navItem)}>
      <Link href={slugFormatter({ slug })}>{label}</Link>
    </li>
  )
}
