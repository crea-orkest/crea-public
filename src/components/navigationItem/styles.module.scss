.navItem {
  padding: 0.5rem 1rem;

  &:hover {
    background-color: rgb(61, 109, 109);
  }
}