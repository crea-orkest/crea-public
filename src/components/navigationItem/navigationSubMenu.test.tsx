import { NavigationItem } from '../navigationItem'
import { render, screen } from '@testing-library/react'

describe('NavigationItem', () => {
  it('should render a link', async () => {
    const { container } = render(
      <NavigationItem slug={'url'} label={'homepage'} />
    )

    expect(screen.getByRole('link', { name: 'homepage' }))
    expect(container).toMatchSnapshot()
  })

  it('should render a link warning', async () => {
    render(<NavigationItem />)

    expect(screen.getByText('No link specified'))
  })

  it('should render a label warning', async () => {
    render(<NavigationItem slug={'test'} />)

    expect(screen.getByText('No label specified'))
  })
})
