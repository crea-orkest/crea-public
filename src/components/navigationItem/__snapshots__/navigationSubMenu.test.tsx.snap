// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NavigationItem should render a link 1`] = `
<div>
  <li
    class="navItem"
  >
    <a
      class="link text-large"
      href="/url"
    >
      homepage
    </a>
  </li>
</div>
`;
